tests/vppinputasync.cpp

```diff
@@ -69,18 +75,47 @@ void VppInputAsync::loop()
    }
 }
 
-bool VppInputAsync::init(const SharedPtr<VppInput>& input, uint32_t queueSize)
+bool VppInputAsync::preloadFrames(uint32_t startupSize)
 {
-    m_input = input;
-    m_queueSize = queueSize;
-    if (pthread_create(&m_thread, NULL, start, this)) {
-        ERROR("create thread failed");
-        return false;
+    uint32_t size;
+    for (size = 0; size < startupSize; size++) {
+        SharedPtr<VideoFrame> frame;
+        if (!m_input->read(frame))
+            break;
+         m_queue.push_back(frame);
+    }
+    if (!size) {
+        ERROR("can't read file from input");
```

Comment:
If the input frame cannot be read, should the function "return false" at this point so that it will not fall thru to the next 'if' statement; unless you intend for the function to fill with duplicate frames regardless of pass/fail.